### PR TITLE
Dynamic Resizing Based on VRAM

### PR DESCRIPTION
**Pull request by wladradchenko**
Added Dynamic Resizing Based on VRAM if option use_limit_device_resolution in cfg is True. Also keep max size by resolution and smaller side will be less than resolution. Dynamic Resizing Based on VRAM from [Issue 79](https://github.com/williamyang1991/Rerender_A_Video/issues/79)

### CODE REVIEW

File: src/video_util.py [hunk 1]
```diff
@@ -98,3 +109,19 @@ def crop_func(frame):
...
+def vram_limit_device_resolution(resolution, device="cuda"):
+    # get max limit target size
+    gpu_vram = torch.cuda.get_device_properties(device).total_memory / (1024 ** 3)
+    # table of gpu memory limit
+    gpu_table = {24: 1280, 18: 1024, 14: 768, 10: 640, 8: 576, 7: 512, 6: 448, 5: 320, 4: 192, 0: 0}
+    # get user resize for gpu
+    device_resolution = max(val for key, val in gpu_table.items() if key <= gpu_vram)
+    print(f"Limit VRAM is {gpu_vram} Gb and size {device_resolution}.")
+    if gpu_vram < 4:
+        print(f"Small VRAM to use GPU. Will be use config resolution")
```
Comment (SingleZombie) [inline src/video_util.py:123]: typo

**Review by SingleZombie — COMMENTED** [review]
LGTM. Could you let me modify the PR and add some docs to README?

**Comment by wladradchenko** [comment]
Yes, of course

**Comment by SingleZombie** [comment]
> Yes, of course . I have checked "Allow edits by maintainers " or do I need to do something yet?

Thank you. That's enough.

**Pull request merged**